casper/src/main/rholang/MakePoS.rho

```diff
@@ -61,6 +62,57 @@ contract @"MakePoS"(@purse, @minimumBond, @maximumBond, @initBonds, return) = {
       }
     } |
     
+    contract feeDistribution(@n, @joiningFee, return) = {
+      new rangeCh, feeDistFunc, feeDistCh, sum, nCh, totalCh in {
+        contract sum(@x, @y, return) = { return!(x + y) } |
+        contract feeDistFunc(@k, return) = { return!((2 * joiningFee * (n + 1 - k)) / (n * (n + 1))) } |
+        @("ListOps", "range")!(1, n + 1, *rangeCh) |
```

Comment:
Yes, that's right. Fortunately we have a ton of unit tests that rely on those sorts of contracts working, so it will not be possible for it to merge until we make them all use unforgable names and the registry. This will be a fair bit of work to refactor them all (and I suppose more work is created every time I make a PR that uses them), but it will need to be done at some point.